### ISSUE TITLE
Cluster instability with high amount of streams

### ISSUE THREAD

**Issue by trevex** [Graylog2/graylog2-server#4471]
I deployed a graylog-cluster on kubernetes in our university cluster, so that all users have easy access to their logs. There is an integration service, that creates new streams for every kubernetes namespace and makes sure permissions are set, so that users (via LDAP) only see their own subset of streams.

There are approximately _200 streams_.

For unknown reasons even after redeploying everything the following error notifications is triggered every few seconds (it disappears and reappears every few seconds):
`There was no master Graylog server node detected in the cluster`
The msg in/out is also unreliable and nodes disappear and reappear. 

Nevertheless the basic functionality is still there, but very slow and unreliable.

The cluster is composed as follows:

1x Graylog Master (8GB Memory)  [receives no input from logs!]
3x Graylog Node (12GB Memory each)
(4x Elasticsearch Master, 3x Elasticsearch Client, 5x Elasticsearch Data)

The Memory limits are JVM settings. Currently graylog gets as much CPU and memory as it needs.
Currently its not near max memory or even using that much CPU resources.

## Expected Behavior
Paging for the list of streams and cluster stability.

## Current Behavior
Loading the list of streams takes forever as Administrator without paging.
Cluster is very unstable and unreliable.

## Possible Solution
I don't know the intrinsic, but  there seems to be a problem with high amount of streams and the work that is created by it. 

## Steps to Reproduce (for bugs)
1. Setup graylog cluster
2. Input about 200msg/s-600msg/s
3. Create 200 Streams filtering those messages

## Context
Basically this is a multi-tenant situation, where there are a lot of permissions, streams and users.

## Your Environment
Containers running on kubernetes

* Graylog Version: graylog2/server:2.4.0-1
* Elasticsearch Version: quay.io/pires/docker-elasticsearch-kubernetes:5.6.0
* MongoDB Version: mongo:3.4.10
* Operating System: Linux
* Browser version: Firefox 57.0.4

## P.S.

If you need more detailed metrics, logs or I should add anything let me know.

**Comment by jalogisch** [comment]
how did you scale your mongodb?

**Comment by trevex** [comment]
Currently just one instance with as many resources as it needs.

Also recored a short gif when idling on the nodes view to illustrate the effects: https://giphy.com/gifs/xULW8tTaLvjzve2V3O/fullscreen

**Comment by trevex** [comment]
UPDATE:
The instabilities only occur during and after the initial batch jobs sets up all the streams. After a few hours the cluster is suddenly stable, before that the IO/CPU/memory consumption is fairly high.

This is therefore certainly an edge case occurring from high amount of api calls, specifically stream creation.

**Comment by jalogisch** [comment]
Unfortunately the issue is not reproducible in the latest release. Please re-open this issue if it still occurs, ideally with a set of reproducible steps. 

Thanks!
